src/uct/ib/mlx5/rc/rc_mlx5_devx.c

```diff
@@ -243,6 +243,15 @@ uct_rc_mlx5_devx_init_rx_common(uct_rc_mlx5_iface_common_t *iface,
 
     iface->rx.srq.db = &iface->rx.srq.devx.dbrec->db[MLX5_RCV_DBR];
 
+    if (iface->config.srq_topo == UCT_RC_MLX5_SRQ_TOPO_AUTO) {
+        ddp_enabled            = (md->dp_ordering_cap.rc ==
+                                  UCT_IB_MLX5_DP_ORDERING_OOO_ALL) ||
+                                 (md->dp_ordering_cap.dc ==
+                                  UCT_IB_MLX5_DP_ORDERING_OOO_ALL);
```

Comment:
can we do this according to the actual transport? maybe pass the ddp_enabled value from caller?